File: Javascript-basics/problem3.js
var largest = 1;
var number = 600851475143 ;
var lock;
var i = 2;
while(i != 1000000){
    i = 2;
    lock = false;
    while (i < 1000000 && lock === false){
        if (number % i === 0){
            number = number / i;
            if (largest < i){
                largest = i;
                lock = true;
            }
        }
        i++;
    }
}

console.log(largest);

File: Ruby basics/problem3.rb
largest = 1
number = 600851475143
i = 0
lock = false
while (i != 1000000)
  i = 2
  lock = false
  while (i < 1000000 && lock == false)
    if (number % i == 0)
      number = number / i
      if (largest < i)
        largest = i
        lock = true
      end
    end
    i = i.next
  end
end